zero_bin/common/src/prover_state/persistence.rs

```diff
@@ -277,11 +279,28 @@ fn prover_to_disk(
     Ok(())
 }
 
-/// If we're running in the cargo workspace, then always use the `circuits`
-/// directory that lives in `tools/`. Otherwise, just use `circuits` in the
-/// current directory.
-fn relative_circuit_dir_path() -> String {
-    env::var(CARGO_WORKSPACE_DIR_ENV)
-        .map(|p| format!("{}/{}", p, CIRCUITS_DIR))
-        .unwrap_or_else(|_| CIRCUITS_DIR.to_string())
+fn circuit_dir() -> String {
+    // Guaranteed to be set by the binary if not set by the user.
+    std::env::var(ZK_EVM_CACHE_DIR_ENV).unwrap_or_else(|_| {
+        format!(
+            "expected the env var \"{}\" to be set",
+            ZK_EVM_CACHE_DIR_ENV
+        )
+    })
+}
```

Comment:
This is a bit confusing.
The documentation `Guaranteed to be set` implies this won't error.
But then we unwrap with a replacement String that isn't actually one, and looks instead like an `expect()` error message, mentioning it actually wasn't set.

Reply:
Oh my bad! That `format!` should be a `panic`.